### PR TITLE
Add check that next focusable list item belongs to the same list

### PR DESCRIPTION
**Pull request by oelhanafey**
[DE53222](https://rally1.rallydev.com/#/?detail=/defect/699633940503&fdp=true)

When navigating a list with arrow keys, the list item will try to focus on the next/previous list item sibling. If that list item is not focusable, it will skip the list item and focus the next focusable element.

The list does not check that the next focusable belongs to the same list. If that non-focusable element is at the start or end of the list, you are able to arrow outside of the list and not able to get back into it.

This PR adds a check to confirm that the next focusable element belongs to the same list as the currently focused element.

### CODE REVIEW

File: components/list/list-item-generic-layout.js [hunk 1]
```diff
@@ -442,6 +444,11 @@ class ListItemGenericLayout extends RtlMixin(LitElement) {
 			this.shadowRoot.querySelector(`.d2l-cell[data-cell-num="${this._cellNum}"]`);
 	}
 
+	_isContainedInList(node1, node2) {
+		const list = findComposedAncestor(node1, node => node.tagName === 'D2L-LIST');
```
[the remark this answers is not in the document]
Comment (oelhanafey) [reply components/list/list-item-generic-layout.js]: Implemented your suggestion and tested in various nested list cases. It seems to be working as expected

File: components/list/list-item-generic-layout.js [hunk 2]
```diff
@@ -301,7 +302,9 @@ class ListItemGenericLayout extends RtlMixin(LitElement) {
 			if (!listItem._tryFocus()) {
 				// ultimate fallback to generic method for getting next/previous focusable
 				const nextFocusable = previous ? getPreviousFocusable(listItem) : getNextFocusable(listItem);
-				if (nextFocusable) nextFocusable.focus();
+				if (nextFocusable && this._isContainedInSameList(curListItem, nextFocusable)) {
```
Comment (dbatiste) [inline components/list/list-item-generic-layout.js:305]: Maybe we should name this `_isContainedInSameRootList` to make it clearer the scope of he check?

**Review by dbatiste — APPROVED** [review]
Looks good!... maybe just a different name for that internal function, but not a huge deal.

File: components/list/list-item-generic-layout.js (hunk 2 above)
Comment (oelhanafey) [reply components/list/list-item-generic-layout.js:305]: Yep, I like that 👍

**Review by dbatiste — APPROVED** [review]

**Pull request merged**